### PR TITLE
removing memchache mentions in comments

### PR DESCRIPTION
**Pull request by rebeccahhh**
##### SUMMARY

just removing all mentions of memcached in comments throughout the codebase. In some cases this is replaced with the appropriate term for redis-caching.

##### ISSUE TYPE

 - Docs Pull Request 

##### COMPONENT NAME

 - API

##### AWX VERSION

```
awx: 12.0.0
```

### CODE REVIEW

File: docs/tasks.md [hunk 1]
```diff
@@ -263,9 +263,9 @@ This entire process enables a response of a `202 Accepted`, where instead of wai
...
+"Handle Setting Changes" provides the solution!  This "fanout" task (_i.e._, all nodes execute it) makes it so that there is a single source of truth even within a clustered system. Whenever a database setting is accessed, and that setting's name is not present in `redis-cache`, it grabs the setting from the database and then populates it in the applicable node's cache.  When any database setting gets altered, all of the `redis-cache` servers on each node needs to "forget" the value that they previously retained. By deleting the setting in `redis-cache` on all nodes with the use of this task, we assure that the next time it is accessed, the database will be consulted for the most up-to-date value.
```
Comment (rebeccahhh) [inline docs/tasks.md:268]: @chrismeyersfsu I particularly want you to take a peek at this section and validate that I am accurate. :)

File: awx/main/tests/conftest.py [hunk 2]
```diff
@@ -131,8 +131,8 @@ def delete(self, key):
 
 def pytest_runtest_teardown(item, nextitem):
     # clear Django cache at the end of every test ran
-    # NOTE: this should not be memcache, see test_cache in test_env.py
-    # this is a local test cache, so we want every test to start with empty cache
+    # NOTE: this should not be memcache (as it is depricated), nor should it be redis.
```
Comment (beeankha) [inline awx/main/tests/conftest.py]: depricated ---> deprecated

File: awx/main/tests/conftest.py [hunk 3]
```diff
@@ -131,8 +131,8 @@ def delete(self, key):
...
+    # This is a local test cache, so we want every test to start with empty cache
```
Comment (beeankha) [inline awx/main/tests/conftest.py]: nitpick: "with _an_ empty cache"

**Review by beeankha — APPROVED** [review]
Pointed out a couple of minor things, otherwise looks great!  👍

File: awx/plugins/inventory/ec2.py [hunk 4]
```diff
@@ -1201,7 +1201,7 @@ def add_elasticache_cluster(self, cluster, region):
 
         # Select the best destination address
         if 'ConfigurationEndpoint' in cluster and cluster['ConfigurationEndpoint']:
-            # Memcached cluster
+            # Redis cluster
```
Comment (wenottingham) [inline awx/plugins/inventory/ec2.py]: this (and below)  is actually pulling memcached clusters from AWS in inventory, it's fine to keep.

@AlanCoding is deleting this shortly anyway.

File: awx/conf/settings.py [hunk 5]
```diff
@@ -31,18 +31,18 @@
 # Store a special value to indicate when a setting is not set in the database.
 SETTING_CACHE_NOTSET = '___notset___'
 
-# Cannot store None in memcached; use a special value instead to indicate None.
+# Cannot store None in cache; use a special value instead to indicate None.
 # If the special value for None is the same as the "not set" value, then a value
 # of None will be equivalent to the setting not being set (and will raise an
 # AttributeError if there is no other default defined).
 # SETTING_CACHE_NONE = '___none___'
 SETTING_CACHE_NONE = SETTING_CACHE_NOTSET
```
Comment (ryanpetrello) [inline awx/conf/settings.py:39]: I'm betting that we don't actually _need_ to do this sort of thing anymore - it's possible that redis and its Python driver have support for storing `None`, `[]`, and `{}`.  That said, this could be a really hairy bit of code to tease apart, and I'm kind of scared to touch it since it's not really hurting anything keeping it this way.

tl;dr if it ain't broke, don't fix it

File: awx/conf/tests/unit/test_settings.py [hunk 6]
```diff
@@ -66,15 +66,6 @@ def test_unregistered_setting(settings):
     assert settings.cache.get('DEBUG') is None
 
 
-def test_cached_settings_unicode_is_auto_decoded(settings):
-    # https://github.com/linsomniac/python-memcached/issues/79
-    # https://github.com/linsomniac/python-memcached/blob/288c159720eebcdf667727a859ef341f1e908308/memcache.py#L961
```
Comment (ryanpetrello) [inline awx/conf/tests/unit/test_settings.py:71]: God I hated this test 👍

File: awx/conf/tests/unit/test_settings.py [hunk 7]
```diff
@@ -254,31 +245,6 @@ def test_setting_from_db(settings, mocker):
...
-def test_setting_from_db_with_unicode(settings, mocker, encrypted):
-    settings.registry.register(
-        'AWX_SOME_SETTING',
-        field_class=fields.CharField,
-        category=_('System'),
-        category_slug='system',
-        default='DEFAULT',
-        encrypted=encrypted
-    )
-    # this simulates a bug in python-memcached; see https://github.com/linsomniac/python-memcached/issues/79
```
Comment (ryanpetrello) [inline awx/conf/tests/unit/test_settings.py:267]: ...and it's ugly cousin.

File: awx/plugins/inventory/ec2.py (hunk 4 above)
Comment (ryanpetrello) [reply awx/plugins/inventory/ec2.py]: Yep, let's leave this file in `awx/plugins` untouched in this PR, it's going away in the near future anyways (https://github.com/ansible/awx/pull/6911).

**Review by ryanpetrello — APPROVED** [review]
A few minor comments, looks great.

File: tools/memcached/.dir_placeholder [hunk 8]
```diff
@@ -1 +1 @@
-This dir must pre-exist and be owned by the user you are launching awx dev env as. If the dir does not exist before launching the awx dev environment then docker will create the dir and it will be owned by root. Since we start our awx dev environment with user: ${CURRENT_UID} the memcached container will be unable to create a socket file in a directory owned by root.
+This dir must pre-exist and be owned by the user you are launching awx dev env as. If the dir does not exist before launching the awx dev environment then docker will create the dir and it will be owned by root. Since we start our awx dev environment with user: ${CURRENT_UID} Redis will be unable to create a socket file in a directory owned by root.
```
Comment (ryanpetrello) [inline tools/memcached/.dir_placeholder]: Actually, I think we can just entirely `git rm tools/memcached` my understanding is that this directory only existed as a place to mount the unix socket file that memcached was listening on cc @chrismeyersfsu

**Comment by rebeccahhh** [comment]
squashed to one commit because this is an appropriate time to use squashing, I should have addressed all the comments, I am pleased to :wave: goodbye to memcached soon :)
